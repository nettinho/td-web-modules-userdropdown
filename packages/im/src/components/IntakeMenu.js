import React from "react";
import { compose } from "redux";
import PropTypes from "prop-types";
import { Icon, Menu } from "semantic-ui-react";
import { FormattedMessage } from "react-intl";
import { withRouter } from "react-router-dom";
import { ActiveRoute } from "@truedat/core/components/ActiveRoute";
import AdminRole from "@truedat/auth/components/AdminRole";
import EngineMenuItem from "../engines/components/EngineMenuItem";
import engineRoutes from "../engines/routes";

export const IntakeMenu = ({ location }) => (
  <Menu.Item
    header
    name="intakes"
    active={ActiveRoute(location, engineRoutes.ENGINES)}
  >
    <Icon name="upload" />
    <FormattedMessage id="navigation.intakes" default="añsdjfañlskdjf" />
    <Menu.Menu>
      <EngineMenuItem />
    </Menu.Menu>
  </Menu.Item>
);

IntakeMenu.propTypes = {
  location: PropTypes.object
};

export default compose(
  withRouter,
  AdminRole
)(IntakeMenu);
