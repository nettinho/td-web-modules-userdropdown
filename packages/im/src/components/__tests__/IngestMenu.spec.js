import React from "react";
import { shallowWithIntl } from "intl-stub";
import { IntakeMenu } from "../IntakeMenu";

describe("<IntakeMenu />", () => {
  it("matches the latest snapshot", () => {
    const location = { pathname: "/" };
    const props = { location };
    const wrapper = shallowWithIntl(<IntakeMenu {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
