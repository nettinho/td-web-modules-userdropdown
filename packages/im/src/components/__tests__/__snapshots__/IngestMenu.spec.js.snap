// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntakeMenu /> matches the latest snapshot 1`] = `
<MenuItem
  active={false}
  header={true}
  name="intakes"
>
  <Icon
    as="i"
    name="upload"
  />
  <FormattedMessage
    default="añsdjfañlskdjf"
    id="navigation.intakes"
    values={Object {}}
  />
  <MenuMenu>
    <withRouter(Connect(EngineMenuItem)) />
  </MenuMenu>
</MenuItem>
`;
